### PR TITLE
Detect vbox/vmware/kvm on Windows and speed up runs

### PR DESCRIPTION
**Pull request by tas50**
Fetch WMI data within Ruby vs. shelling out to Powershell.  This should
be WAY faster.

### CODE REVIEW

**Comment by tas50** [comment]
Going to rework the specs for virtualization so hold off on this one for now

**Comment by btm** [comment]
@tas50 this is cool. you should finish it. :)

also hyperv. See #647, although that's about linux guests.

**Comment by tas50** [comment]
I need to sit down and beef up my rspec skillset to test this.

**Comment by btm** [comment]
VMware Fusion: https://gist.github.com/btm/f044387275ffa277d745

**Comment by tas50** [comment]
@btm This is ready now.  Also it speeds up the Ohai runs by 6% and works if PS isn't installed.

File: spec/unit/plugins/windows/virtualization_spec.rb [hunk 1]
```diff
@@ -18,47 +20,159 @@
...
-SMBIOSBIOSVersion : 10.2.0 (28956) rev 0
-Manufacturer      : Parallels Software International Inc.
-Name              : Default System BIOS
-SerialNumber      : Parallels-92 05 B4 56 97 11 4F FA B1 95 1A FF 8E F9 DD CE
-Version           : PRLS   - 1
-      BIOS
-      shellout = double("shellout")
-      allow(shellout).to receive(:stdout).and_return(bios)
-      allow(plugin).to receive(:shell_out).with('powershell.exe "Get-WmiObject -Class Win32_BIOS"').and_return(shellout)
+  context 'when running on vmware fusion' do
```
Comment (btm) [inline spec/unit/plugins/windows/virtualization_spec.rb]: vmware fusion or parallels? it looks like two tests got combined by accident here?

Comment (tas50) [inline spec/unit/plugins/windows/virtualization_spec.rb]: oh looks like I did a select and delete on part of the block before I did the commit :(  Let me push it with the correct test block.

**Comment by btm** [comment]
woot :+1: presuming it all passes. I was hoping you'd shoot for 7 or 8%, but 6% will do. ;)

**Comment by thommay** [comment]
:+1: thanks Tim!

**Pull request merged**